src/lib/storage/segment_sorted_search.hpp

```diff
@@ -17,101 +17,155 @@ class SortedSegmentSearch {
  public:
   SortedSegmentSearch(IteratorType begin, IteratorType end, const OrderByMode& order_by,
                       const PredicateCondition& predicate, const SearchValueType& search_value)
-      : _begin{begin},
-        _end{end},
+      : _original_begin{begin},
+        _original_end{end},
+        _begin_offset{0},
+        _end_offset{ChunkOffset(std::distance(begin, end))},
         _order_by{order_by},
         _predicate{predicate},
         _search_value{search_value},
         _is_ascending{order_by == OrderByMode::Ascending || order_by == OrderByMode::AscendingNullsLast},
         _is_nulls_first{order_by == OrderByMode::Ascending || order_by == OrderByMode::Descending} {}
 
  private:
+  // TODO(cmfcmf): This is only needed until #1512 is fixed and we can use std::lower_bound.
```

Comment:
Please add a reference to this in the issue. The way the issue reads right now, it being fixed might not lead to this being changed.